power_grid_model_io_native_c/power_grid_model_io_native_c/src/vnf_pgm_converter.cpp

```diff
@@ -19,14 +19,21 @@ struct PGM_IO_VnfConverter : public PgmVnfConverter {
 };
 
 // TODO(Laurynas-Jagutis) add call_with_catch for these functions.
-PGM_IO_VnfConverter* PGM_IO_create_vnf_converter(const PGM_IO_Handle* /*handle*/, char* file_buffer) {
-    auto* converter = new PGM_IO_VnfConverter(file_buffer);
-    parse_vnf_file_wrapper(converter);
-    return converter;
+PGM_IO_VnfConverter* PGM_IO_create_vnf_converter(PGM_IO_Handle* handle, char* file_buffer,
+                                                 PGM_IO_Idx experimental_features) {
+    return call_with_catch(
+        handle,
+        [file_buffer, experimental_features] {
+            auto* converter = new PGM_IO_VnfConverter(file_buffer, nullptr, experimental_features);
```

Comment:
why `nullptr`?

Reply:
This needs to be replaced with a PGM_WritableDataset. Here I have another question, at the moment the PGM_WritableDataset is defined in power_grid_model_c/basics.h. But I can't import it from there. right(I already tried including the file, mission was not succesful)? Do I create my own struct in power_grid_model_io_native_c/basics.h and work with it or do I approach in another way?

Reply:
you can create your own opaque struct in `power_grid_model_io_native_c/basics.h` and then where you define the implementation, you can use `using PGM_IO_WritableDataset = power_grid_model::WritableDataset;` (included from `power_grid_model/auxiliary/dataset.h`) or similar

Reply:
but why do you need `WritableDataset`? i thought we would go for the serialization from string to string approach? in that case, you would need to use the `WritableDataset` in the last step of the converter implementation, but not across the C API.